### PR TITLE
core: add pagination and update example

### PR DESCRIPTION
**Pull request by HayesGordon**
This is a draft pull request.

There are a few things to consider:
- Better ways to handle/store the pagination within the bloc
- Deprecating `queryEnrichedActivities` in favor of `queryPaginatedEnrichedActivities`
- There are a few places that will benefit from logs, marked as TODO
- Expanding on the tests

We will also need similar logic for `ReactionListCore` and wherever else pagination is needed.

NOTE: There is a bug in the childReactions, where `ownChildren` is not being retrieved/stored on a reaction. This is a separate issue outside of this pull request, however, the updated example illustrates the problem.

### CODE REVIEW

File: packages/stream_feed_flutter_core/lib/src/bloc/feed_bloc.dart [hunk 1]
```diff
@@ -60,30 +61,53 @@ class GenericFeedBloc<A, Ob, T, Or> extends Equatable {
...
       [String? kind]) {
     return reactionsManager.getStream(activityId, kind);
   }
 
+  /// The current activities list.
+  List<GenericEnrichedActivity<A, Ob, T, Or>>? getActivities(
+          String feedGroup) =>
+      activitiesManager.getActivities(feedGroup);
+
+  /// The current of grouped activities list.
```
Comment (HayesGordon) [inline packages/stream_feed_flutter_core/lib/src/bloc/feed_bloc.dart:94]: /// The current grouped (aggregated) activities list

File: packages/stream_feed_flutter_core/lib/src/bloc/feed_bloc.dart [hunk 2]
```diff
@@ -469,8 +497,348 @@ class GenericFeedBloc<A, Ob, T, Or> extends Equatable {
...
+      } else {
+        activitiesManager.addError(feedGroup, e, stk);
+      }
+    }
+  }
+
+  /// Queries the activities stream and stores the pagination results.
+  ///
+  /// Unique activities will be stored and can be retrieved by calling
+  /// [getActivities].
```
Comment (HayesGordon) [inline packages/stream_feed_flutter_core/lib/src/bloc/feed_bloc.dart:637]: should be [getGroupedActivities]

File: packages/stream_feed_flutter_core/lib/src/bloc/feed_bloc.dart [hunk 3]
```diff
@@ -469,8 +497,348 @@ class GenericFeedBloc<A, Ob, T, Or> extends Equatable {
...
+  ///
+  /// To load more enriched activities, see [loadMoreEnrichedActivities], or
```
Comment (HayesGordon) [inline packages/stream_feed_flutter_core/lib/src/bloc/feed_bloc.dart:639]: should be [loadMoreGroupedActivities]

**Comment by rlee1990** [comment]
@HayesGordon @sachaarbonel Any update on this? I really need this.

**Pull request merged**